### DISCUSSION TITLE
Openweathermap: split free (one-call) and paying

### DISCUSSION THREAD

**Discussion by koying** [home-assistant/architecture#671] [General]
Openweathermap has basically 2 API's nowadays: The free one called "One Call" for the free tiers, and the paying one.
Those upstream API have diverged quite a bit nowadays, but we still manage both under the same integration.

Of interest is that One-Call, the free one, returns both daily and hourly forecasts in a single API call, which is not reflected in the current integration to keep an unified one, so something that would require a single call now requires 2.

This to explore the opportunity to split the integration into 2: The free one would be basically rebuilt from the ground up while the paying one (which I assume would be *very* lightly used) would basically stay as it is, but removing the "onecall-*" modes.

**Comment by allenporter** [comment]
Which of these are you suggesting?
- make two integrations
- restructure the code internally to make it easier to manage two APIs within the same integration?
 
The best practice is to keep a single integration and manage everything inside of it rather than making two separate integrations for two APIs. There are a few examples of doing this (e.g. `nest`) and it is a little more difficult to manage the code structure, but worthwhile in the long run from the user perspective.

I'd say the most difficult parts with code structure are config flow, since there can only be one per integration, but good tests can help keep it manageable.

**Comment by frenck** [comment]
This architecture issue is old, stale, or possibly obsolete. Things changed a lot over the years.

For that reason, I'm going to close this discussion for now to clean up a little.

../Frenck
